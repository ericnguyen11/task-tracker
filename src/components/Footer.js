import { Link } from "react-router-dom";

const Footer = () => {
  return (
    <footer>
      <Link className="footer-link" to="/about">
        About
      </Link>
    </footer>
  );
};

export default Footer;
