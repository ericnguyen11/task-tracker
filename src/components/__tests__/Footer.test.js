import { MemoryRouter } from "react-router-dom";

import { mount, shallow } from "enzyme";

import Footer from "../Footer.js";

it("renders without crashing", () => {
  shallow(<Footer />);
});

it("matches snapshot", () => {
  const wrapper = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
});

it("includes link to about", () => {
  const wrapper = mount(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );
  expect(wrapper.find(".footer-link").at(0).props().to).toBe("/about");
});
