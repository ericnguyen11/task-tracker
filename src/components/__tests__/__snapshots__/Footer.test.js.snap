// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<footer>
  <Link
    className="footer-link"
    to="/about"
  >
    About
  </Link>
</footer>
`;
